File: composeApp/src/desktopMain/kotlin/fr/thomasbernard03/androidtools/domain/usecases/application/InstallApplicationUseCase.kt
package fr.thomasbernard03.androidtools.domain.usecases.application

import fr.thomasbernard03.androidtools.domain.models.InstallApplicationResult
import fr.thomasbernard03.androidtools.domain.repositories.ApplicationRepository
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import org.koin.java.KoinJavaComponent.inject

class InstallApplicationUseCase {
    private val applicationRepository: ApplicationRepository by inject(ApplicationRepository::class.java)

    suspend operator fun invoke(path : String, onStatusChanged : (InstallApplicationResult) -> Unit) = withContext(Dispatchers.IO) {
        onStatusChanged(InstallApplicationResult.Loading)
        val finalStatus = applicationRepository.installApplication(path)
        onStatusChanged(finalStatus)
    }
}

File: lib/shared/domain/usecases/install_application_usecase.dart
import 'package:android_tools/shared/domain/repositories/application_repository.dart';

class InstallApplicationUsecase {
  final ApplicationRepository _applicationRepository;
  InstallApplicationUsecase(this._applicationRepository);
  Future<void> call(String apkPath, String deviceId) {
    return _applicationRepository.intallApplication(apkPath, deviceId);
  }
}
